server/src/main/java/org/opensearch/index/shard/ReplicationGroup.java

```diff
@@ -75,7 +86,13 @@ public ReplicationGroup(
                 skippedShards.add(shard);
             } else {
                 if (trackedAllocationIds.contains(shard.allocationId().getId())) {
-                    replicationTargets.add(shard);
+                    replicationTargets.add(
+                        new ReplicationAwareShardRouting(
+                            remoteTranslogEnabled,
+                            localTranslogAllocationIds.contains(shard.allocationId().getId()),
```

Comment:
Based on the `assert` that is added at the beginning of this constructor, if `remoteTranslogEnabled==true`, the above statement will always be true. Is this understanding correct?

Reply:
the assert at the beginning of constructor is when `remoteTranslogEnabled==false`. When `remoteTranslogEnabled==false`, then all shards have local translogs. However, when  `remoteTranslogEnabled==true`, there will be atleast the primary shard which has local translog and a relocating primary in case of primary-primary relocation.